packages/twenty-server/src/engine/api/graphql/workspace-schema-builder/services/type-mapper.service.ts

```diff
@@ -61,7 +63,9 @@ export class TypeMapperService {
       fieldMetadataType === FieldMetadataType.NUMBER &&
```

Comment:
I'm more and more convinced that we should implement a FieldMetadata class for each type now that we have settings. We can then type check with a guard and we wouldn't have to do type assertion everywhere

Reply:
Open to discussion ofc if you are not convinced.

One thing that I don't like are the settings themselves. We wanted to have `precision` to have something more scalable than subtypes for number. But since we are now introducing `isBigInt`, should I instead use an enum:
`INT`, `FLOAT`, `BIG_INT` ?

cc @charlesBochet